crates/turborepo-lib/src/run/mod.rs

```diff
@@ -60,16 +60,41 @@ use crate::{
     turbo_json::TurboJson,
 };
 
-#[derive(Debug)]
 pub struct Run {
     base: CommandBase,
     processes: ProcessManager,
+    opts: Opts,
+    api_auth: Option<APIAuth>,
 }
 
 impl Run {
-    pub fn new(base: CommandBase) -> Self {
+    pub fn new(base: CommandBase, api_auth: Option<APIAuth>) -> Result<Self, Error> {
         let processes = ProcessManager::new();
-        Self { base, processes }
+        let mut opts: Opts = base.args().try_into()?;
+        let config = base.config()?;
+        let is_linked = turborepo_api_client::is_linked(&api_auth);
+        if !is_linked {
+            opts.cache_opts.skip_remote = true;
+        } else if let Some(enabled) = config.enabled {
+            // We're linked, but if the user has explicitly enabled or disabled, use that
+            // value
+            opts.cache_opts.skip_remote = !enabled;
+        }
+        // Note that we don't currently use the team_id value here. In the future, we
+        // should probably verify that we only use the signature value when the
+        // configured team_id matches the final resolved team_id.
+        let unused_remote_cache_opts_team_id = config.team_id().unwrap_or_default();
```

Comment:
I don't love that we're flattening a None to an empty String here. It's not particular idiomatic to Rust.

Reply:
This is just moving around some existing logic. I'll see how invasive it is to switch to an `Option`, but I'll also note that that's not the biggest issue here. The real problem is we have an unused team id that we pull from the config.

Reply:
Ended up not being too bad. It's now an `Option<String>` called `unused_team_id`.